### ISSUE TITLE
Versioning and releases

### ISSUE THREAD

**Issue by daniel-j-h** [mapbox/robosat#81]
Eventually we want robosat to be a library with a thin user-interface (command line tools) on top of it: https://github.com/mapbox/robosat/issues/9. One step in that direction is making proper releases and versioning the code.

We should
- use semantic versioning for releases
- look into PyPI for publishing releases

Semantic versioning is especially required to communicate with users when we change our mode architectures and when loading old `.pth` weight files will no longer work. We should avoid that in the first place but if we have to make changes we should be explicit about it.

The releasing playbook could look like the following
- Check Travis is green
- Tag the release git tag vx.y.z -a on master, git push origin vx.y.z
- Make Github Release for tag
- Push release onto PyPI

**Comment by daniel-j-h** [comment]
We should follow https://packaging.python.org/tutorials/packaging-projects/

and look into publishing packages to PyPI directly through Travis for every release we do.

**Comment by daniel-j-h** [comment]
With https://github.com/mapbox/robosat/pull/87 we are building Docker images now for latest master and for all tags:

https://hub.docker.com/r/mapbox/robosat/tags/

I just tagged `v0.0.1`.

From now on we need to respect semantic versioning, e.g. every time we break the `.pth` format because we change something in our model Python classes we need to bump the major version number.

The process is as follows:
- make pull request
- wait on Travis
- merge into master
- `git tag vx.y.z -a`

we then will automatically get Docker images for this tag (and latest for all commits in master).

cc @karenzshea
